include/seqan3/alignment/pairwise/detail/type_traits.hpp

```diff
@@ -117,4 +126,24 @@ struct alignment_configuration_traits
         static_cast<original_score_type>(1u << (sizeof_bits<original_score_type> - 1));
 };
 
+//------------------------------------------------------------------------------
+// alignment_function_traits
+//------------------------------------------------------------------------------
+
+/*!\brief A traits class to provide a uniform access to the properties of the wrapped alignment algorithm.
+ * \ingroup pairwise_alignment
+ *
+ * \tparam function_t The type of the std::function object that stores the alignment algorithm as the target.
+ */
+template <typename function_t>
```

Comment:
Template requirements are missing.
This seems like a "generic" usable for any aligment algoirhtm functor so readable error messages would be nice?

Reply:
Yes and no. It is not completely trivial and the STL does not provide a suitable type trait to check if something is a function object. So I wanted to skip this 😁  But I make another PR that adds is_function_or_is_function_object to core/utility

Reply:
Not sure if that is so easy. Anyway. I would make it another PR and add this if this already merged.